recipe/build.sh

```diff
@@ -1,12 +1,20 @@
-# explicitly link mpi, openblas
-export LDFLAGS="-L$PREFIX/lib -lmpi -llapack -lblas $LDFLAGS"
```

Comment:
Why is `-llapack -lblas` removed?

Reply:
It's just a workaround for the ppc64le_mpich build, which does not link to libmpi for some reason.

Reply:
What does mpi got to do with blas and lapack?

Reply:
It's that the CMake MPI configure test on ppc64le_mpich fails without explicit having `-lmpi` in `LDFLAGS`. It has nothing to do with blas and lapack. I'm rerunning with a better debug output to show he exact error message.

Reply:
here is the error message without `export LDFLAGS="-L$PREFIX/lib -lmpi $LDFLAGS"` on linux ppc64le mpich:

```
+LDFLAGS=-Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib
INFO: activate-gxx_linux-ppc64le.sh made the following environmental changes:
+CXXFLAGS=-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -mcpu=power8 -mtune=power8 -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/hypre-2.21.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix
+CXX_FOR_BUILD=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++
+CXX=$BUILD_PREFIX/bin/powerpc64le-conda-linux-gnu-c++
+DEBUG_CXXFLAGS=-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -mcpu=power8 -mtune=power8 -ftree-vectorize -fPIC -fstack-protector-all -fno-plt -Og -g -Wall -Wextra -fvar-tracking-assignments -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/hypre-2.21.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix
+GXX=$BUILD_PREFIX/bin/powerpc64le-conda-linux-gnu-g++
+ [[ mpich == \o\p\e\n\m\p\i ]]
+ cd src/cmbuild
+ cmake -DCMAKE_AR=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-ar -DCMAKE_CXX_COMPILER_AR=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-gcc-ar -DCMAKE_C_COMPILER_AR=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-gcc-ar -DCMAKE_RANLIB=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-ranlib -DCMAKE_CXX_COMPILER_RANLIB=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-gcc-ranlib -DCMAKE_C_COMPILER_RANLIB=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-gcc-ranlib -DCMAKE_LINKER=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-ld -DCMAKE_STRIP=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/bin/powerpc64le-conda-linux-gnu-strip -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY '-DCMAKE_FIND_ROOT_PATH=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place;/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/powerpc64le-conda-linux-gnu/sysroot' -DCMAKE_INSTALL_PREFIX=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_PROCESSOR=ppc64le -DHYPRE_SHARED=ON -DBUILD_SHARED_LIBS=ON -DHYPRE_USING_HYPRE_BLAS=OFF -DHYPRE_USING_HYPRE_LAPACK=OFF -DHYPRE_USING_FEI=OFF -DHYPRE_INSTALL_PREFIX=/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place ..
-- The C compiler identification is GNU 9.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: $BUILD_PREFIX/bin/powerpc64le-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Could NOT find MPI_C (missing: MPI_C_LIB_NAMES MPI_C_WORKS) 
CMake Error at /home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MPI (missing: MPI_C_FOUND)
Call Stack (most recent call first):
  /home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /home/conda/feedstock_root/build_artifacts/hypre_1622060681468/_build_env/share/cmake-3.20/Modules/FindMPI.cmake:1742 (find_package_handle_standard_args)
  CMakeLists.txt:378 (find_package)


-- Configuring incomplete, errors occurred!
See also "$SRC_DIR/src/cmbuild/CMakeFiles/CMakeOutput.log".
See also "$SRC_DIR/src/cmbuild/CMakeFiles/CMakeError.log".
+ cat CMakeFiles/CMakeError.log
The MPI test test_mpi for C in mode normal failed to compile with the following output:
Change Dir: $SRC_DIR/src/cmbuild/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_38278/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_38278.dir/build.make CMakeFiles/cmTC_38278.dir/build
gmake[1]: Entering directory `$SRC_DIR/src/cmbuild/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_38278.dir/test_mpi.c.o
$BUILD_PREFIX/bin/powerpc64le-conda-linux-gnu-cc   -mcpu=power8 -mtune=power8 -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/hypre-2.21.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -o CMakeFiles/cmTC_38278.dir/test_mpi.c.o -c $BUILD_PREFIX/share/cmake-3.20/Modules/FindMPI/test_mpi.c
Linking C executable cmTC_38278
$BUILD_PREFIX/bin/cmake -E cmake_link_script CMakeFiles/cmTC_38278.dir/link.txt --verbose=1
$BUILD_PREFIX/bin/powerpc64le-conda-linux-gnu-cc -mcpu=power8 -mtune=power8 -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/hypre-2.21.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib  CMakeFiles/cmTC_38278.dir/test_mpi.c.o -o cmTC_38278 
+ exit 1
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: CMakeFiles/cmTC_38278.dir/test_mpi.c.o: in function `main':
test_mpi.c:(.text.startup+0x48): undefined reference to `MPI_Init'
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: test_mpi.c:(.text.startup+0x4c): undefined reference to `MPI_Init'
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: test_mpi.c:(.text.startup+0x50): undefined reference to `MPI_Init'
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: test_mpi.c:(.text.startup+0x54): undefined reference to `MPI_Init'
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: test_mpi.c:(.text.startup+0x5c): undefined reference to `MPI_Finalize'
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: test_mpi.c:(.text.startup+0x60): undefined reference to `MPI_Finalize'
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: test_mpi.c:(.text.startup+0x64): undefined reference to `MPI_Finalize'
$BUILD_PREFIX/bin/../lib/gcc/powerpc64le-conda-linux-gnu/9.3.0/../../../../powerpc64le-conda-linux-gnu/bin/ld: test_mpi.c:(.text.startup+0x68): undefined reference to `MPI_Finalize'
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_38278] Error 1
gmake[1]: Leaving directory `$SRC_DIR/src/cmbuild/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_38278/fast] Error 2



Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 481, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 470, in execute
    outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 3068, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 2191, in build
    utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/utils.py", line 412, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/conda_build/utils.py", line 392, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/home/conda/feedstock_root/build_artifacts/hypre_1622060681468/work/conda_build.sh']' returned non-zero exit status 1.
##[error]Bash exited with code '1'.

```

Reply:
I get that we need `-lmpi` for linux ppc64le. I'm wondering why you've removed `-llapack -lblas`.

Reply:
> I get that we need `-lmpi` for linux ppc64le. I'm wondering why you've removed `-llapack -lblas`.

It didn't seem to make a difference. I will restore the original LDFLAGS line.

Reply:
Restored the original flags.